enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,921 @@
+
+Provide an automated mechanism to upgrade a cluster without requiring an image
+registry server.
+
+## Motivation
+
+All these stories are in the context of disconnected clusters with limited
+resources, both in the cluster itself and in the surrounding environment:
+
+- The cluster is not connected to the Internet, or the bandwidth is very limited.
+- It isn't possible to bring up additional machines, even temporarily.
+- The resources of the cluster are limited, in particular in terms of CPU and memory.
+- The technicians responsible for performing the upgrade have no OpenShift knowledge.
+
+These clusters are usually installed at the customer site by a partner engineer
+collaborating with customer technicians.
+
+Eventually, the cluster will need to be upgraded, and then the technicians will
+need tools that make the process as simple as possible, ideally requiring no
+OpenShift knowledge.
+
+When OpenShift knowledge is required the technician performing the upgrade will
+have the support of the team of engineers that prepared and vetted the upgrade
+bundle.
+
+### User Stories
+
+#### Prepare an upgrade bundle
+
+As an engineer working in a partner's factory, I want to be able to assemble an
+upgrade bundle that contains all the artifacts (container images and metadata)
+needed to upgrade the OpenShift cluster and my own applications. I want to hand
+over this upgrade bundle and the documentation explaining how to use it to the
+technicians that will perform the upgrade in a suitable media, for example a USB
+stick.
+
+#### Include custom images in the bundle
+
+As an engineer working in a partner's factory I want to be able to include in
+the upgrade bundle custom images for workloads specific to the customer.
+
+#### Explicitly allow vetted upgrade bundle
+
+As an engineer managing a cluster I want be able to explicitly approve the use
+of an upgrade bundle, so that only the bundle that I tested and vetted will be
+applied to the cluster.
+
+#### Upgrade a single-node cluster
+
+As a technician with little or no OpenShift knowledge I want to be able to
+upgrade a single-node cluster using the upgrade bundle and its documentation. I
+can't bring up any additional infrastructure at the cluster site, in particular
+I can't bring up an image registry server, neither outside of the cluster nor
+inside. I want to plug the USB stick provided by the engineer in the node and
+have the rest of the process performed automatically.
+
+#### Upgrade a multi-node cluster
+
+As a technician with little or no OpenShift knowledge I want to be able to
+upgrade a multi-node cluster as well. This is the same story than for
+single-node clusters but for multi-node clusters. The difference is that I don't
+want to plug the USB stick to all the nodes of the cluster, instead I want to
+plug it to only one of the nodes (selected randomly) and have the rest of the
+process performed automatically, included the propagation of the bundle contents
+to all the nodes.
```

Comment:
Looks like the two stories above could be really merged into one, as long as the node where the USB stick is plugged doesn't matter